zbench.zig

```diff
@@ -219,46 +225,162 @@ pub const Benchmark = struct {
 /// It takes a pointer to a Benchmark object.
 pub const BenchFunc = fn (*Benchmark) void;
 
-/// BenchmarkResult stores the result of a single benchmark.
-/// It includes the name and the total duration of the benchmark.
+/// BenchmarkResult stores the resulting computed metrics/statistics from a benchmark
 pub const BenchmarkResult = struct {
-    /// Name of the benchmark.
+    const Self = @This();
+    const Color = c.Color;
+
+    /// Name of the benchmark
     name: []const u8,
-    /// Total duration of the benchmark in nanoseconds.
-    duration: u64,
+    /// 75th, 99th and 99.5th percentiles of the recorded durations. They represent the timings below
+    /// which 75%, 99% and 99.5% of the other measurments would lie, respectively, when timings
+    /// are sorted in increasing order.
+    percs: Benchmark.Percentiles,
+    /// The average (more precisely arithmetic mean) of the recorded durations
+    avg_duration: usize,
+    /// The standard-deviation of the recorded durations (an estimate for the average *deviation* from
+    /// the average duration).
+    std_duration: usize,
+    /// The minimum among the recorded durations
+    min_duration: usize,
+    /// The maximum among the recorded durations
+    max_duration: usize,
+    /// The total amount of operations (or runs) performed of the benchmark
+    total_operations: usize,
+    /// Total time for all the operations (or runs) of the benchmark combined
+    total_time: usize,
+
+    /// Pretty-prints the name of the benchmark
+    /// writer: Type that has the associated method print (for example std.io.getStdOut.writer())
+    pub fn prettyPrintName(self: Self, writer: anytype, color: Color) !void {
```

Comment:
nit: From my point of view, it would be great if these functions would move into the `util` folder.